compute_endpoint/globus_compute_endpoint/strategies/base.py

```diff
@@ -40,60 +43,49 @@ class BaseStrategy:
     from a duplicate logger being added by the thread.
     """
 
-    def __init__(self, *args, threshold: int = 20, interval: float = 5.0):
+    def __init__(self, *args, interval: float = 5.0):
         """Initialize the flowcontrol object.
 
         We start the timer thread here
-        Parameters
-        ----------
-        - threshold (int) : Tasks after which the callback is triggered
-        - interval (int) : seconds after which timer expires
 
+        :param interval: how often (in seconds) to invoke self.strategy
         """
-        self.interchange = None
-        self.threshold = threshold
+        self.interchange: t.Optional[GlobusComputeEngine] = None
         self.interval = interval
 
         self.cb_args = args
-        self.callback = self.strategize
         self._handle = None
-        self._event_count = 0
-        self._event_buffer: t.List[t.Any] = []
-        self._wake_up_time = time.time() + 1
         self._kill_event = threading.Event()
         self._thread: t.Optional[threading.Thread] = None
 
-    def start(self, interchange):
+    def start(self, engine: "GlobusComputeEngine"):
         """Actually start the strategy
-        Parameters
-        ----------
-        interchange:
-         globus_compute_endpoint.executors.high_throughput.interchange.Interchange
-            Interchange to bind the strategy to
+
+        :param engine: Engine to which this strategy is bound
         """
         # This thread is created here to ensure a new thread is created whenever start
         # is called. This is to avoid errors from tests reusing strategy objects which
         # would attempt to restart stopped threads.
         self._thread = threading.Thread(
-            target=self._wake_up_timer, args=(self._kill_event,), name="Base-Strategy"
+            target=self._wake_up_timer, name="Base-Strategy", daemon=True
         )
-        self._thread.daemon = True
-        self.interchange = interchange
-        if hasattr(interchange, "provider"):
+        self.interchange = engine
+        if hasattr(engine, "provider"):
             log.debug(
                 "Strategy bounds-> init:{}, min:{}, max:{}".format(
-                    interchange.provider.init_blocks,
-                    interchange.provider.min_blocks,
-                    interchange.provider.max_blocks,
+                    engine.provider.init_blocks,
+                    engine.provider.min_blocks,
+                    engine.provider.max_blocks,
                 )
             )
         self._thread.start()
 
+    @abstractmethod
     def strategize(self, *args, **kwargs):
-        """Strategize is called everytime the threshold or the interval is hit"""
-        log.debug(f"Strategize called with {args} {kwargs}")
+        """Strategize is called everytime he interval is hit"""
```

Comment:
Typo `he -> the`

Reply:
I'll get this in a follow-up PR.  Want to merge this (on my day off!).